include/aws/s3/private/s3_client_impl.h

```diff
@@ -100,10 +100,6 @@ struct aws_s3_client_vtable {
     struct aws_s3_meta_request *(
         *meta_request_factory)(struct aws_s3_client *client, const struct aws_s3_meta_request_options *options);
 
-    void (*push_meta_request)(struct aws_s3_client *client, struct aws_s3_meta_request *meta_request);
```

Comment:
Push and remove of meta requests for processing is now handled internally by the client.  (Before, the meta request would have to manage when to re-push itself.)